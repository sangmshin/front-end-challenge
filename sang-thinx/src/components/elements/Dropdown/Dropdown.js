import React, { useState } from 'react';
import { array, string } from 'prop-types';
import cx from 'classnames';
import {
  DropdownButton,
  List
} from 'components/elements';
import styles from './Dropdown.module.scss';

const Dropdown = ({
  className,
  items,
  type
}) => {
  const [selectedItem, selectItem] = useState(null);
  const [isListOpen, toggleList] = useState(false);

  const handleDropdown = item => {
    selectItem(item);
    toggleList(false);
  };

  return (
    <div className={cx(styles.root, className)}>
      <DropdownButton
        isOpen={isListOpen}
        selectedItem={selectedItem}
        onClick={() => toggleList(isListOpen ? false : true)}
        type={type}
      />
      {isListOpen &&
        <List
          items={items}
          onSelectItem={handleDropdown}
          selectedItem={selectedItem}
        />
      }
    </div>
  )
}

Dropdown.propTypes = {
  className: string,
  items: array.isRequired,
  type: string.isRequired
}

Dropdown.defaultProps = {
  className: ''
}

export default Dropdown;