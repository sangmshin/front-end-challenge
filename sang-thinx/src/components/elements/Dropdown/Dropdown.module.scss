.root {
  position: relative;
  border: 1px solid black;
  padding: 0 7px;
  height: 30px;
  display: flex;
  align-items: center;
}