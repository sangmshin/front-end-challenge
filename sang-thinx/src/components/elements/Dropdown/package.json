{
  "name": "Dropdown",
  "main": "./Dropdown.js"
}
