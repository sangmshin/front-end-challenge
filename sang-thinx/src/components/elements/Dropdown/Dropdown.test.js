import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Dropdown } from './Dropdown';

const props = {
  items: [
    'New York',
    'Los Angeles',
    'Chicago'
  ],
  type: 'City'
};

const component = shallow(<Dropdown {...props} />);

describe('Component :: Dropdown', () => {
  describe('#render', () => {
    it('should match snapshot', () => {
      expect(toJson(component)).toMatchSnapshot();
    });
  });

  describe('#prop availability: type', () => {
    it('should have 1 "DropdownButton" component and it should have correct "type" prop', () => {
      const DROPDOWN_TYPE = props.type;
      expect(component.find('DropdownButton').length).toBe(1);
      expect(component.find('DropdownButton').prop('type')).toBe(DROPDOWN_TYPE);
    });
  });
});
