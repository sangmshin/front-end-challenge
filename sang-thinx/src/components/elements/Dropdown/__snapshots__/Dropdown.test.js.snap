// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component :: Dropdown #render should match snapshot 1`] = `
<div
  className="root"
>
  <DropdownButton
    onClick={[Function]}
    selectedItem={null}
    type="City"
  />
</div>
`;
